### ISSUE TITLE
`DTZ005`: false positive when `replace(...)` before `astimezone()`

### ISSUE THREAD

**Issue by trim21** [astral-sh/ruff#15998]
### Description

```python
import datetime


print("update at", datetime.datetime.now().replace(microsecond=0).astimezone())
```

https://play.ruff.rs/ca289032-db85-4067-bbf2-e77e59d90365

**Comment by ayushbaweja** [comment]
Should DTZ005 only allow replace() method chains before astimezone(), or should it allow any method/attribute chain as long as the object remains a datetime.datetime before calling astimezone()?

For example, would both of these be valid?
```
datetime.now().replace(hour=12).astimezone()  
datetime.now().other_method().foo.bar().astimezone()  # Using other methods/attributes that return datetime object
```

**Comment by trim21** [comment]
I think only method return datatime can be valid here？ replace is the only method on datatime return a datatime.
